src/nominatim_api/results.py

```diff
@@ -225,23 +208,30 @@ def lon(self) -> float:
         """
         return self.centroid[0]
 
+    @property
+    def display_name(self) -> Optional[str]:
+        """ Dynamically compute the display name for the result place
+            and, if available, its address information..
+        """
+        # No longer sets the locale name, want to do that explicitly in format.py
+        # using the formatter object
+        if self.address_rows:  # if this is true we need additional processing
+            label_parts: List[str] = []
+
+            for line in self.address_rows:  # assume locale_name is set by external formatter
+                if line.isaddress and line.names and line.local_name:  # checks if it is an address
+                    if not label_parts or label_parts[-1] != line.local_name:
+                        label_parts.append(line.local_name)
+            return ', '.join(label_parts)
+        return self.locale_name
```

Comment:
So this function currently works only, if localization was called and the `locale_name` property set. Let's make it work in all cases.

For each part the name should be the first of:
1. `locale_name`
2. `names['name']`
3. the first name found: `next(iter(names.values()))`

In line 226 there might be an addtional fallback to `housenumber` needed.